batavia/builtins/dir.js

```diff
@@ -11,7 +12,12 @@ function dir(args, kwargs) {
         throw new exceptions.TypeError.$pyclass('dir() expected exactly 1 argument (' + args.length + ' given)')
     }
     if (args.length === 0) {
-        return "['__builtins__', '__cached__', '__doc__', '__file__', '__loader__', '__name__', '__package__', '__spec__', 'f', 'x']"
+        // Python 3.7 added __annotations__ for use with dataclass
+        if (!version.earlier(3.7)) {
+            return "['__annotations__', '__builtins__', '__cached__', '__doc__', '__file__', '__loader__', '__name__', '__package__', '__spec__', 'f', 'x']"
+        } else {
+            return "['__builtins__', '__cached__', '__doc__', '__file__', '__loader__', '__name__', '__package__', '__spec__', 'f', 'x']"
+        }
```

Comment:
This may be a subject for a follow on PR - but is this return value correct? dir() returns a list, not a string... this looks like something that will *pass* tests, but only because the output format of the string value is indistinguisable from the rendered version of a list.

Reply:
Ha! It does pass the tests, even though it is wrong in every case. It's a big change so I'll fill an issue for it and we can fix it separately. Funny how it is so easy to follow the format and never think about how it isn't right.